embed-gutenberg-block-google-maps.php

```diff
@@ -14,168 +14,171 @@
-    $APIkey = get_option('pantheon_google_map_block_api_key');
-
-    // Don't output anything if there is no API key
-    if (null === $APIkey || empty( $APIkey )) {
-        return;
-    }
-
-    // Exapnd all the atributes into separate variables
-    foreach($attributes as $key=>$value) {
-        ${$key} = $value; 
-    }
-
-    // URL encode the location for Google Maps
-    $location = urlencode ( $location );
-
-    // Set the API url based to embed or static maps based on the interactive setting
-    $apiURL = ( $interactive ) ? "https://www.google.com/maps/embed/v1/place?key=${APIkey}&q=${location}&zoom=${zoom}&maptype=${mapType}" : "https://maps.googleapis.com/maps/api/staticmap?center=${location}&zoom=${zoom}&size=${maxWidth}x${maxHeight}&maptype=${mapType}&key=${APIkey}";
-    
-    // Check status code of apiURL
-    $ch = curl_init($apiURL);
-    curl_setopt($ch, CURLOPT_HEADER, true);
-    curl_setopt($ch, CURLOPT_NOBODY, true);
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER,1);
-    curl_setopt($ch, CURLOPT_TIMEOUT,10);
-    $output = curl_exec($ch);
-    $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-    curl_close($ch);
-
-    // Don't output anything if the response from Google Maps isn't a 200
-    if( $httpcode !== 200 ){
-        return;
-    }
-
-    // Set the appropriate CSS class names
-    $classNames = ( $interactive ) ? "wp-block-pantheon-google-map interactive ratio$aspectRatio" : "wp-block-pantheon-google-map";
-    
-    // Create the output
-    $output = "<div class='$classNames'><div class='map'>";
-    // If the map is interactive show the iframe
-    if( $interactive ){
-        $output .= "<iframe width='100%' height='100%' frameborder='0' style='border:0' src='$apiURL' allowfullscreen></iframe>";
-    // Otherwise use the static API
-    } else {
-        $output .= "<img src='$apiURL' />";
-    }
-    $output .= '</div></div>';
-
-    // Return the output
-    return $output;
+function render_gutenberg_map_embed_block( $attributes ) {
+
+	// Get the API key.
+	$api_key = get_option( 'pantheon_google_map_block_api_key' );
+
+	// Don't output anything if there is no API key.
+	if ( null === $api_key || empty( $api_key ) ) {
+		return;
+	}
+
+	// Expand all the attributes into separate variables.
+	foreach ( $attributes as $key => $value ) {
+		${$key} = $value;
+	}
+
+	// URL encode the location for Google Maps.
+	$location = rawurlencode( $location );
+
+	// Set the API URL based to embed or static maps based on the interactive setting.
+	$api_url = ( $interactive ) ? "https://www.google.com/maps/embed/v1/place?key=${api_key}&q=${location}&zoom=${zoom}&maptype=${mapType}" : "https://maps.googleapis.com/maps/api/staticmap?center=${location}&zoom=${zoom}&size=${maxWidth}x${maxHeight}&maptype=${mapType}&key=${api_key}"; /* phpcs:ignore */
+
+	// Check status code of the API.
+	// phpcs:disable WordPress.WP.AlternativeFunctions
+	$ch = curl_init( $api_url );
+	curl_setopt( $ch, CURLOPT_HEADER, true );
```

Comment:
Looks like this is the option that prevents the use of `WP_HTTP`, the remaining options are defaults/can be set via the API.

Part interesting fact/part desire to check if it's essential.